### PR TITLE
support lightning model

### PR DESCRIPTION
**Pull request by yexinyinancy**
### 4. How to test?
- [ ] N/A
- [ ] Unit test
- [ ] Application test
- [ ] Document test
- [ ] ...

### 5. New dependencies

- [ ] New Python dependencies
       - Dependency1 
       - Dependency2
       - ...
- [ ] New Java/Scala dependencies and their license
       - Dependency1 and license1
       - Dependency2 and license2
       - ...

### CODE REVIEW

**Comment by yexinyinancy** [comment]
Currently support:
1. pytorch-lightning model with function ``forward``, ``cross_entropy_loss``, ``training_step``, ``loss``, ``validation_step`` and ``configure_optimizers``.
2. The return type of ``training_step`` can be loss or dictionary.
3. The return type of ``configure_optimizers`` can be optimizer, scheduler and dictionary.
4. pytorch_lightning callbacks.
5. The input data type should be xshards.

**Comment by yexinyinancy** [comment]
Current issue:
1. If the input feature has only one column, x in ``training_step`` is a list, we need to use x[0] to get the actual data and put it into the model.
2. We cannot support all the function in pytorch_lightning callbacks.
